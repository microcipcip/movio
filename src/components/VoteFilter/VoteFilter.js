import React from 'react'

const VoteFilter = ({ ratingValue, setVoteFilter, isActive }) => (
  <div onClick={() => setVoteFilter(ratingValue)}>{ratingValue}</div>
)

export default VoteFilter
