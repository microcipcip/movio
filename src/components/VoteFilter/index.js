import Index from './VoteFilter'

export default Index
