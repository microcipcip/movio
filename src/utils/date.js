export const addZero = date => ('0' + date).slice(-2)

export const formatMovieDate = movieDate => {
  const d = new Date(movieDate)
  return `${addZero(d.getDate())} ${addZero(
    d.getMonth() + 1
  )} ${d.getFullYear()}`
}
