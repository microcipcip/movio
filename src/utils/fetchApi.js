// wrapper for the fetch API
// because it doesn't throw errors when response status is 400
import { TMDB_API_KEY, TMDB_API_URL, TMDB_API_LANG } from 'config/index'

const parseJSON = response =>
  new Promise(resolve =>
    response.json().then(json =>
      resolve({
        status: response.status,
        ok: response.ok,
        json,
      })
    )
  )

export default url => {
  const queryOrAmpersand = url.includes('?') ? '&' : '?'
  const query = `${TMDB_API_URL}${url}${queryOrAmpersand}api_key=${TMDB_API_KEY}&language=${TMDB_API_LANG}`

  return new Promise((resolve, reject) =>
    fetch(query)
      .then(parseJSON)
      .then(response => {
        if (response.ok) {
          return resolve(response.json)
        }
        // extract the error from the server's json
        return reject(response.json.meta.error)
      })
      .catch(err =>
        reject({
          networkError: err.message,
        })
      )
  )
}
