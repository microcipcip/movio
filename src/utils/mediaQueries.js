import { css } from 'styled-components'

// Breakpoint
export const mediaWidthRule = rule =>
  `${{ up: 'min', down: 'max' }[rule] || 'min'}-width`

export const ruleTemplate = (rule, width) => `(${rule}: ${width}px)`
export const mediaTemplate = rules => `@media only screen and ${rules}`

export const getSmallestMedia = breakpoints => {
  const keys = Object.keys(breakpoints)

  return Number(
    keys.reduce((acc, key) => {
      if (acc) {
        return acc > breakpoints[key] ? breakpoints[key] : acc
      }
      return breakpoints[key]
    }, undefined)
  )
}

export const getNextMedia = (breakpoints, width) => {
  const keys = Object.keys(breakpoints)

  const nextBreakpoint = Number(
    keys.reduce((acc, key) => {
      const current = breakpoints[key]
      if (acc && width - current <= 0) {
        return width - acc > Math.abs(width - current) || width - acc === 0
          ? current
          : acc
      }
      return getSmallestMedia(breakpoints)
    }, undefined)
  )

  if (nextBreakpoint === width) {
    throw new Error(
      `The breakpoint of value ${nextBreakpoint} is the heighst there is, did you mean to use "[media].up"?`
    )
  }
  return nextBreakpoint
}

export const mediaRules = (breakpoints, widthKey, rule, boundKey) => {
  const width = breakpoints[widthKey]
  const bound = breakpoints[boundKey]
  let baseWidthRule = mediaWidthRule(rule)
  let boudWidthRule

  let baseRule = ruleTemplate(baseWidthRule, width)
  let boundRule

  if (bound && width) {
    // Get correct rule based on width relative to bound
    baseWidthRule = mediaWidthRule(bound <= width ? 'down' : 'up')
    boudWidthRule = mediaWidthRule(bound <= width ? 'up' : 'down')

    baseRule = ruleTemplate(
      mediaWidthRule(bound <= width ? 'down' : 'up'),
      width
    )
    boundRule = ruleTemplate(boudWidthRule, bound)
    return [].concat([baseRule], bound ? [boundRule] : []).join(' and ')
  }

  if (!bound && rule === 'only') {
    // Get correct rule based on width relative to bound
    boudWidthRule = mediaWidthRule(bound <= width ? 'up' : 'down')
    boundRule = ruleTemplate(boudWidthRule, getNextMedia(breakpoints, width))
  }

  return [].concat([baseRule], boundRule ? [boundRule] : []).join(' and ')
}

export const getMixin = (breakpoints, key, rule = 'up') => bound => (
  ...args
) => css`
  ${mediaTemplate(mediaRules(breakpoints, key, rule, bound))} {
    ${css(...args)}
  }
`

export const getMediaShorthands = (breakpoints, rule) =>
  Object.keys(breakpoints).reduce(
    (acc, key) => ({
      ...acc,
      [key]: getMixin(breakpoints, key, rule)(),
    }),
    {}
  )

export const getMedia = breakpoints => {
  const mediasUp = getMediaShorthands(breakpoints, 'up')
  const list = Object.keys(breakpoints)

  return {
    ...mediasUp,
    up: widthKey => getMixin(breakpoints, widthKey, 'up')(),
    down: widthKey => getMixin(breakpoints, widthKey, 'down')(),
    only: (widthKey, boundKey) =>
      getMixin(breakpoints, widthKey, 'only')(boundKey),
    list,
  }
}

export default getMedia
