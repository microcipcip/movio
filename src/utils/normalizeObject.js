export default (acc, cur) => {
  acc[cur.id] = cur
  return acc
}
