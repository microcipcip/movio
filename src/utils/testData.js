export const genreList = {
  1: {
    id: 1,
    name: 'Action',
    isChecked: false,
  },
  2: {
    id: 2,
    name: 'Adventure',
    isChecked: false,
  },
  3: {
    id: 3,
    name: 'Animation',
    isChecked: false,
  },
  4: {
    id: 4,
    name: 'Comedy',
    isChecked: false,
  },
  5: {
    id: 5,
    name: 'Crime',
    isChecked: false,
  },
}

export const movieList = [
  {
    id: 1,
    vote_average: 6.5,
    title: 'Venom',
    popularity: 423.154,
    poster_path: '/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
    genre_ids: [1, 2],
    release_date: '2018-10-03',
  },
  {
    id: 2,
    vote_average: 7,
    title: 'Fantastic Beasts: The Crimes of Grindelwald',
    popularity: 220.606,
    poster_path: '/uyJgTzAsp3Za2TaPiZt2yaKYRIR.jpg',
    genre_ids: [3, 4],
    release_date: '2018-11-14',
  },
  {
    id: 3,
    vote_average: 8.2,
    title: 'Bohemian Rhapsody',
    popularity: 211.565,
    poster_path: '/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
    genre_ids: [5],
    release_date: '2018-10-24',
  },
  {
    id: 4,
    vote_average: 5.9,
    title: 'Mortal Engines',
    popularity: 201.549,
    poster_path: '/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg',
    genre_ids: [1],
    release_date: '2018-12-06',
  },
  {
    id: 5,
    vote_average: 6,
    title: 'Robin Hood',
    popularity: 171.519,
    poster_path: '/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg',
    genre_ids: [2],
    backdrop_path: '/AuA50D7T7S7OEVcGo0ZKaMhERn0.jpg',
    release_date: '2018-11-20',
  },
  {
    id: 6,
    vote_average: 5.9,
    title: 'The Seven Deadly Sins: Prisoners of the Sky',
    popularity: 150.182,
    poster_path: '/r6pPUVUKU5eIpYj4oEzidk5ZibB.jpg',
    genre_ids: [3],
    release_date: '2018-08-18',
  },
  {
    id: 7,
    vote_average: 7.5,
    title: 'Ralph Breaks the Internet',
    popularity: 141.193,
    poster_path: '/m110vLaDDOCca4hfOcS5mK5cDke.jpg',
    genre_ids: [4, 5],
    release_date: '2018-11-20',
  },
]
