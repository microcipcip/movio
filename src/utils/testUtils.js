import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import { genreList, movieList } from 'utils/testData'
import * as s from 'styles/vars'

import CinemaContext, { cinemaContextState } from 'components/App/CinemaContext'

const customRender = (node, data = {}) => {
  return render(
    <CinemaContext.Provider value={Object.assign(cinemaContextState, data)}>
      <ThemeProvider theme={s.theme.dark}>{node}</ThemeProvider>
    </CinemaContext.Provider>
  )
}

const domToString = el => {
  const clnEl = el.cloneNode(true)
  const tmp = document.createElement('div')
  tmp.appendChild(clnEl)
  return tmp.innerHTML
}

// re-export everything
export * from 'react-testing-library'
export { customRender, domToString, genreList, movieList }
