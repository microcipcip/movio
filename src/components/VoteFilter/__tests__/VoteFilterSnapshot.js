import React from 'react'
import { customRender, movieList } from 'utils/testUtils'
import VoteFilter from '../VoteFilter'

it('Should create snapshot', async () => {
  const { container } = customRender(<VoteFilter />, { movieList })
  expect(container).toMatchSnapshot()
})
