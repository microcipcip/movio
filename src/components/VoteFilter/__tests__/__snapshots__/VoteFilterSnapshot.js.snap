// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create snapshot 1`] = `
.c0 {
  display: block;
  min-width: 140px;
  position: relative;
}

.c1 .input-range__track--active {
  background: #1a7eff;
}

.c1 .input-range__slider {
  width: 1.3rem;
  height: 1.3rem;
  margin-top: -0.8rem;
  background-color: #1a7eff;
  border-color: #1a7eff;
}

.c1 .input-range__label-container {
  font-size: 1rem;
}

.c1 .input-range__label--value {
  top: -2.4rem;
}

.c1 .input-range__label--value .input-range__label-container {
  left: 0;
}

.c1 .input-range__label--min,
.c1 .input-range__label--max {
  bottom: -1.6rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        aria-disabled="false"
        class="input-range"
      >
        <span
          class="input-range__label input-range__label--min"
        >
          <span
            class="input-range__label-container"
          >
            0
          </span>
        </span>
        <div
          class="input-range__track input-range__track--background"
        >
          <div
            class="input-range__track input-range__track--active"
            style="left: 0%; width: 30%;"
          />
          <span
            class="input-range__slider-container"
            style="position: absolute; left: 30%;"
          >
            <span
              class="input-range__label input-range__label--value"
            >
              <span
                class="input-range__label-container"
              >
                3
              </span>
            </span>
            <div
              aria-valuemax="10"
              aria-valuemin="0"
              aria-valuenow="3"
              class="input-range__slider"
              draggable="false"
              role="slider"
              tabindex="0"
            />
          </span>
        </div>
        <span
          class="input-range__label input-range__label--max"
        >
          <span
            class="input-range__label-container"
          >
            10
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
